bib/internal/progress/progress.go

```diff
@@ -0,0 +1,296 @@
+	}
+	ppb.poolStarted = true
+	return nil
+}
+
+func (ppb *termProgressBar) Stop() (err error) {
+	// pb.Stop() will deadlock if it was not started before
+	if ppb.poolStarted {
+		err = ppb.pool.Stop()
+	}
+	ppb.poolStarted = false
+	return err
+}
+
+type plainProgressBar struct {
+	w io.Writer
+}
+
+// NewPlainProgressBar starts a new "plain" progressbar that will just
+// prints message but does not show any progress.
+func NewPlainProgressBar() (ProgressBar, error) {
+	np := &plainProgressBar{w: osStderr}
+	return np, nil
+}
+
+func (np *plainProgressBar) SetPulseMsg(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, msg, args...)
+}
+
+func (np *plainProgressBar) SetMessage(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, msg, args...)
+}
+
+func (np *plainProgressBar) Start() (err error) {
+	return nil
+}
+
+func (np *plainProgressBar) Stop() (err error) {
+	return nil
+}
+
+func (np *plainProgressBar) SetProgress(subLevel int, msg string, done int, total int) error {
+	return nil
+}
+
+type debugProgressBar struct {
+	w io.Writer
+}
+
+// NewDebugProgressBar will create a progressbar aimed to debug the
+// lower level osbuild/images message. It will never clear the screen
+// so "glitches/weird" messages from the lower-layers can be inspected
+// easier.
+func NewDebugProgressBar() (ProgressBar, error) {
+	np := &debugProgressBar{w: osStderr}
+	return np, nil
+}
+
+func (np *debugProgressBar) SetPulseMsg(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, "pulse: ")
+	fmt.Fprintf(np.w, msg, args...)
+	fmt.Fprintf(np.w, "\n")
+}
+
+func (np *debugProgressBar) SetMessage(msg string, args ...interface{}) {
+	fmt.Fprintf(np.w, "msg: ")
+	fmt.Fprintf(np.w, msg, args...)
+	fmt.Fprintf(np.w, "\n")
+}
+
+func (np *debugProgressBar) Start() (err error) {
+	fmt.Fprintf(np.w, "Start progressbar\n")
+	return nil
+}
+
+func (np *debugProgressBar) Stop() (err error) {
+	fmt.Fprintf(np.w, "Stop progressbar\n")
+	return nil
+}
+
+func (np *debugProgressBar) SetProgress(subLevel int, msg string, done int, total int) error {
+	fmt.Fprintf(np.w, "%s[%v / %v] %s", strings.Repeat("  ", subLevel), done, total, msg)
+	fmt.Fprintf(np.w, "\n")
+	return nil
+}
+
+// XXX: merge variant back into images/pkg/osbuild/osbuild-exec.go
+func RunOSBuild(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
+	switch pb.(type) {
+	case *termProgressBar, *debugProgressBar:
+		return runOSBuildNew(pb, manifest, store, outputDirectory, exports, extraEnv)
+	default:
+		return runOSBuildOld(pb, manifest, store, outputDirectory, exports, extraEnv)
+	}
+}
+
+func runOSBuildOld(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
+	_, err := osbuild.RunOSBuild(manifest, store, outputDirectory, exports, nil, extraEnv, false, os.Stderr)
+	return err
+}
+
+func runOSBuildNew(pb ProgressBar, manifest []byte, store, outputDirectory string, exports, extraEnv []string) error {
```

Comment:
I know that this is temporary, but the function name is really bad. :D

Reply:
Uhhh, indeed it is, sorry for that! Fixed now 

/me grabs a brown paperbag.

Reply:
I tweaked the naming, please let me know if you have more suggestions here (naming is hard!)